src/desktop/chat/chatwidget.cpp

```diff
@@ -619,8 +619,9 @@ void ChatWidget::receiveMessage(int sender, int recipient, uint8_t tflags, uint8
 	if(!d->myline->hasFocus() || chatId != d->currentChat)
 		notification::playSound(notification::Event::CHAT);
 
-	if(wasAtEnd)
+	if(wasAtEnd || isValidAlert) {
```

Comment:
The point of an alert is to foist it on the user. For example, mods want to be able to alert users to curable rule breaks. Only if they don't get a reaction, they invade the session. The most common reason for not getting a reaction is "I had the chat closed/scrolled up."

It's probably not the right solution though, since you can still pop out the chat, shove it on a different desktop and then not see alerts anyway. Showing alerts as some kind of unmissable notification balloon would be better. Adding another message type for mod messages would be yet even better, rather than intermixing them with system alerts. We could for example spring up a modal dialog that interrupts everything.

Reply:
Thanks for the explanation.

I have a pending patch that moves system alerts to the netstatus pop-up bubble if the chat window is collapsed (because the chat opening every time the system wanted me to know it was auto-resetting the canvas was annoying), but from what you are describing it does sound like there are certain kinds of messages that need to be presented more obtrusively, but this seems OK for now to me.